datalad/local/subdatasets.py

```diff
@@ -271,11 +290,15 @@ def _get_submodules(ds, paths, fulfilled, recursive, recursion_limit,
         return
     # put in giant for-loop to be able to yield results before completion
     for sm in _parse_git_submodules(ds, paths):
-        if contains and not any(
-                sm['path'] == c or sm['path'] in c.parents for c in contains):
-            # we are not looking for this subds, because it doesn't
-            # match the target path
-            continue
+        contains_hits = []
+        if contains:
+            contains_hits = [
+                c for c in contains if sm['path'] == c or sm['path'] in c.parents
+            ]
+            if not contains_hits:
+                # we are not looking for this subds, because it doesn't
+                # match the target path
+                continue
```

Comment:
So the main price we pay for this added information is not aborting the `for c in contains` iteration early if one of the elements is a hit.  OK.